[aotinductor] Only autotune at compile time when enabled via config

ColinPeppler opened the pull request:
Summary: ---

Test Plan: ci

Differential Revision: D58931057


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @desertfire @chauhang

@diff-train-skip-merge

desertfire commented on torch/_inductor/codegen/wrapper.py:
```diff
@@ -1488,7 +1488,7 @@ def wrap_arg(arg):
                 return str(arg)
             else:
-                return pexpr(V.graph.sizevars.simplify(arg))
+                return self.expr_printer(V.graph.sizevars.simplify(arg))
```
Will this fix the internal model failure you pointed out in D58931057?

desertfire commented on torch/_inductor/compile_fx.py:
```diff
@@ -1300,7 +1300,7 @@ def compile_fx(
             {
                 "cpp_wrapper": False,
-                "triton.autotune_at_compile_time": True,
+                "triton.autotune_at_compile_time": config.triton.autotune_at_compile_time,
```
Can you also update here, https://github.com/pytorch/pytorch/blob/main/torch/_inductor/config.py#L700, to make the default value to 1? I wanted to keep the always on unless it breaks anything beyond the current test coverage. Not sure if your change to wrapper.py fixes the underlying problem. If yes, you can make the default value of `TORCHINDUCTOR_TRITON_AUTOTUNE_AT_COMPILE_TIME` always to 1; otherwise, you can make the default of `TORCHINDUCTOR_TRITON_AUTOTUNE_AT_COMPILE_TIME` different between fbcode and OSS.

ColinPeppler replied:
Making fbcode and OSS default difference since the issue still exists

ColinPeppler committed:
[aotinductor] Only autotune at compile time when enabled via config (#129335)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/129335

 ---

Test Plan: ci

Reviewed By: jingsh, desertfire, houseroad

Differential Revision: D58931057
